File: src/main/java/dark/leech/text/models/Pager.java
package dark.leech.text.models;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Pager {
    private String url;
    private String name;
    private List<Chapter> chapter;
    private String id;
    private boolean completed;

    public Pager(String url, int id) {
        this.url = url;
        this.id = "Q" + id;
    }

    public Pager(String url) {
        this.url = url;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setId(int id) {
        this.id = "P" + id;
    }
}

File: engine/src/commonMain/kotlin/dev/haipham22/leechtext/models/Pager.kt
package dev.haipham22.leechtext.models

/** Một trang pagination (port từ models/Pager.java). */
data class Pager(
    var url: String? = null,
    var name: String? = null,
    var chapter: List<Chapter>? = null,
    var id: String? = null,
    var completed: Boolean = false,
) {
    fun setId(id: Int) {
        this.id = "P$id"
    }
}
